go/vt/tabletserver/query_executor.go

```diff
@@ -107,7 +107,11 @@ func (qre *QueryExecutor) Execute() (reply *mproto.QueryResult) {
 			defer conn.Recycle()
 			reply = qre.execSQL(conn, qre.query, true)
 		default:
-			reply = qre.execDmlAutoCommit()
+			if qre.qe.enableAutoCommit {
```

Comment:
idiomatic go wants you to avoid the else:
if !qre... {
  panic(...)
}
reply = ...